datavec/datavec-python/src/main/java/org/datavec/python/PythonContextManager.java

```diff
@@ -0,0 +1,165 @@
+
+/*******************************************************************************
+ * Copyright (c) 2019 Konduit K.K.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+
+package org.datavec.python;
+
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Emulates multiples interpreters in a single interpreter
+ *
+ * @author Fariz Rahman
+ */
+
+
+public class PythonContextManager {
+
+    private static Set<String> contexts = new HashSet<>();
+    private static boolean init = false;
+    private static String currentContext;
+
+    static {init();}
+    private static void init(){
+        if (init) return;
+        new PythonExecutioner();
+        init = true;
+        currentContext = "main";
+        contexts.add(currentContext);
+    }
+
+    public static void addContext(String contextName){
+        if(!_validateContextName(contextName)){
+            throw new RuntimeException("Invalid context name: " + contextName);
+        }
+        contexts.add(contextName);
+    }
+
+    public static boolean hasContext(String contextName){
+        return contexts.contains(contextName);
+    }
+
+
+    public static boolean _validateContextName(String s) {
```

Comment:
Let's get rid of the pythonisms. Keep private/protected methods without the _ prefix.